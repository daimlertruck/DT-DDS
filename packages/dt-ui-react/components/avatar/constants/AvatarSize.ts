export const AvatarSize = {
  Small: 'small',
  Medium: 'medium',
  Large: 'large',
} as const;

export type AvatarSize = (typeof AvatarSize)[keyof typeof AvatarSize];
