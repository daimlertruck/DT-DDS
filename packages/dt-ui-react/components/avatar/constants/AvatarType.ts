export enum AvatarType {
  Profile = 'profile',
  TeamMember = 'teamMember',
}
