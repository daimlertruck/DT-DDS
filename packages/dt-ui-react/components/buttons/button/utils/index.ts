export * from './buttonVariantColors';
export * from './buttonVariantStyles';
