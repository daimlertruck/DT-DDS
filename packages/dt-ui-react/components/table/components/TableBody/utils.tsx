import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
