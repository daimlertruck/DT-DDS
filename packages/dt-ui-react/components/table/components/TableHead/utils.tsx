import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
