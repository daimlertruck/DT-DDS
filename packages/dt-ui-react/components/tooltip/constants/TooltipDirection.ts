export const OTooltipDirection = {
  Top: 'top',
  Bottom: 'bottom',
  Right: 'right',
  Left: 'left',
} as const;

export type TooltipDirection =
  (typeof OTooltipDirection)[keyof typeof OTooltipDirection];
