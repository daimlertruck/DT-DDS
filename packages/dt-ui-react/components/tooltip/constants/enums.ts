export enum TooltipDirection {
  Top = 'top',
  Bottom = 'bottom',
  Right = 'right',
  Left = 'left',
}
