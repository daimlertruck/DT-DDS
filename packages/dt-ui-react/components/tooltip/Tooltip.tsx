import { ReactNode } from 'react';

import { BaseProps } from '../../types';

import { TooltipDirection } from './constants';
import { ContentStyled, TooltipStyled } from './Tooltip.styled';

export interface TooltipProps extends BaseProps {
  content: ReactNode;
  direction?: TooltipDirection;
  styleContainer?: React.CSSProperties;
}

export const Tooltip = ({
  children,
  content,
  direction = TooltipDirection.Top,
  style,
  styleContainer,
}: TooltipProps) => {
  return (
    <TooltipStyled style={styleContainer}>
      {children}
      {content && (
        <ContentStyled direction={direction} role='tooltip' style={style}>
          {content}
        </ContentStyled>
      )}
    </TooltipStyled>
  );
};
