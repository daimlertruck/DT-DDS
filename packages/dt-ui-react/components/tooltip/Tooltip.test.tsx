import {
  render,
  fireEvent,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import React from 'react';

import { withProviders, withTooltipProvider } from '../../utils';

import { TooltipDirection } from './constants';
import Tooltip from './Tooltip';

describe('<Tooltip /> component', () => {
  const ProvidedTooltip = withProviders(withTooltipProvider(Tooltip));

  it('renders a tooltip correctly', async () => {
    const { getByTestId } = render(
      <ProvidedTooltip>
        Hover me
        <Tooltip.Content direction={TooltipDirection.Top}>
          Some content
        </Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    await waitFor(() => getByTestId('tooltip-content'));

    expect(getByTestId('tooltip-content')).toMatchSnapshot();
  });

  it('renders a tooltip on the right side of the content', async () => {
    const { getByTestId } = render(
      <ProvidedTooltip>
        Hover me
        <Tooltip.Content direction={TooltipDirection.Right}>
          Some content
        </Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    await waitFor(() => getByTestId('tooltip-content'));

    expect(getByTestId('tooltip-content')).toMatchSnapshot();
  });

  it('renders a tooltip on the left side of the content', async () => {
    const { getByTestId } = render(
      <ProvidedTooltip>
        Hover me
        <Tooltip.Content direction={TooltipDirection.Left}>
          Some content
        </Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    await waitFor(() => getByTestId('tooltip-content'));

    expect(getByTestId('tooltip-content')).toMatchSnapshot();
  });

  it('renders a tooltip on the bottom side of the content', async () => {
    const { getByTestId } = render(
      <ProvidedTooltip>
        Hover me
        <Tooltip.Content direction={TooltipDirection.Bottom}>
          Some content
        </Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    await waitFor(() => getByTestId('tooltip-content'));

    expect(getByTestId('tooltip-content')).toMatchSnapshot();
  });
  it('should hide tooltip after the hide delay', async () => {
    jest.useFakeTimers();

    const { getByTestId, queryByTestId } = render(
      <ProvidedTooltip hideDelay={1000}>
        Hover me
        <Tooltip.Content>Some content</Tooltip.Content>
      </ProvidedTooltip>
    );

    fireEvent.mouseEnter(getByTestId('tooltip-container'));

    fireEvent.mouseLeave(getByTestId('tooltip-container'));

    jest.advanceTimersByTime(1000);

    await waitForElementToBeRemoved(() => getByTestId('tooltip-content'));

    expect(queryByTestId('tooltip-content')).not.toBeInTheDocument();
  });
});
