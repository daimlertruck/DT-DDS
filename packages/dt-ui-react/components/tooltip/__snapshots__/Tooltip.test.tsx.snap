// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> component renders a tooltip correctly 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing:normal margin: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 160px;
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #000A25;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.15);
  color: #FFFFFF;
  border-radius: 4px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 6;
  top: -10px;
  left: 0px;
  -webkit-transform: translate(-50%, -100%);
  -moz-transform: translate(-50%, -100%);
  -ms-transform: translate(-50%, -100%);
  transform: translate(-50%, -100%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border-width: 5px;
  border-style: solid;
  border-color: transparent;
}

.emotion-0:after {
  top: 100%;
  left: 50%;
  margin-left: -5px;
  border-top-color: #000A25;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="top"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip on the bottom side of the content 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing:normal margin: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 160px;
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #000A25;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.15);
  color: #FFFFFF;
  border-radius: 4px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 6;
  top: 10px;
  left: 0px;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border-width: 5px;
  border-style: solid;
  border-color: transparent;
}

.emotion-0:after {
  bottom: 100%;
  left: 50%;
  margin-left: -5px;
  border-bottom-color: #000A25;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="bottom"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip on the left side of the content 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing:normal margin: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 160px;
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #000A25;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.15);
  color: #FFFFFF;
  border-radius: 4px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 6;
  top: 0px;
  left: -10px;
  -webkit-transform: translate(-100%, -50%);
  -moz-transform: translate(-100%, -50%);
  -ms-transform: translate(-100%, -50%);
  transform: translate(-100%, -50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border-width: 5px;
  border-style: solid;
  border-color: transparent;
}

.emotion-0:after {
  top: 50%;
  left: 100%;
  margin-top: -5px;
  border-left-color: #000A25;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="left"
  role="tooltip"
>
  Some content
</div>
`;

exports[`<Tooltip /> component renders a tooltip on the right side of the content 1`] = `
.emotion-0 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing:normal margin: 0;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 160px;
  overflow-wrap: break-word;
  padding: 8px;
  background-color: #000A25;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.15);
  color: #FFFFFF;
  border-radius: 4px;
  text-align: left;
  white-space: initial;
  position: absolute;
  z-index: 6;
  top: 0px;
  left: 10px;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-0:after {
  content: "";
  position: absolute;
  border-width: 5px;
  border-style: solid;
  border-color: transparent;
}

.emotion-0:after {
  top: 50%;
  right: 100%;
  margin-top: -5px;
  border-right-color: #000A25;
}

<div
  class="emotion-0 emotion-1"
  data-testid="tooltip-content"
  direction="right"
  role="tooltip"
>
  Some content
</div>
`;
